### DISCUSSION TITLE
Is `any` converter supported officially?

### DISCUSSION THREAD

**Discussion by gh640** [pallets/flask#5208] [Q&A] [answered]
In the Quickstart page, `any` converter of routing is *not* included in "Converter types":

https://flask.palletsprojects.com/en/2.3.x/quickstart/#variable-rules

> Converter types:
> 
> ========== ==========================================
> ``string`` (default) accepts any text without a slash
> ``int``    accepts positive integers
> ``float``  accepts positive floating point values
> ``path``   like ``string`` but also accepts slashes
> ``uuid``   accepts UUID strings
> ========== ==========================================

https://github.com/pallets/flask/blob/cb825687a592709f902f3d320d93987a0546fd28/docs/quickstart.rst?plain=1#L208

In the API page, `any` converter is in the available converters:

https://flask.palletsprojects.com/en/2.3.x/api/#url-route-registrations

> The following converters are available:
> 
> =========== ===============================================
> `string`    accepts any text without a slash (the default)
> `int`       accepts integers
> `float`     like `int` but for floating point values
> `path`      like the default but also accepts slashes
> `any`       matches one of the items provided
> `uuid`      accepts UUID strings
> =========== ===============================================

https://github.com/pallets/flask/blob/cb825687a592709f902f3d320d93987a0546fd28/docs/api.rst?plain=1#L567-L568

Also, `any` is in "Built-in Converters" in Werkzeug documentation:

https://werkzeug.palletsprojects.com/en/2.3.x/routing/#built-in-converters

> `class werkzeug.routing.AnyConverter(map, *items)`
> Matches one of the items provided. Items can either be Python identifiers or strings:
> 
> ```python
> Rule('/<any(about, help, imprint, class, "foo,bar"):page_name>')
> ```

---

It looks like `any` converter in Quickstart was removed in the following commit.

https://github.com/pallets/flask/commit/2592f927a00f39ca089f67144a1e7f200ad213d1

---

`any` converter is useful for me but I have no idea if I should use it or not. Any reason it was removed from Quickstart? Thanks.

**Comment by RonnyPfannschmidt** [comment]
Follow up note, should get the name one-of to make it more clear what's the intent?

**Comment by davidism** [comment, marked as answer]
Yes it's supported by Werkzeug. It looks like the PR that removed it was trying to make the quickstart page less verbose and more focused on basic use cases.

**Reply by gh640** [reply]
@davidism 

Thanks for the answer.

I was unaware that there was `any` converter and I found it via Stack Overflow this time. Currently it's not very clear if the converters listed on the Quickstart page are a subset or all of them and I had to follow the following links to find out all the supported converters:

https://flask.palletsprojects.com/en/2.3.x/quickstart/#variable-rules
↓
`route()`
https://flask.palletsprojects.com/en/2.3.x/api/#flask.Flask.route
↓
`add_url_rule()`
https://flask.palletsprojects.com/en/2.3.x/api/#flask.Flask.add_url_rule
↓
URL Route Registrations
https://flask.palletsprojects.com/en/2.3.x/api/#url-route-registrations

For example, I think it would be easier to newbies like me to add a note to the Quickstart page that explains that the converter list on the Quickstart page is only a subset of the list and that all converters can be found at https://flask.palletsprojects.com/en/2.3.x/api/#url-route-registrations .

(Moreover, it may be better to direct people who want to know how to use the converters with parameters to the Werkzeug's documentation.)

Anyway, my question has been answered. Thanks!
